src/noc/containers/noc-field/utils/noc-validators.ts

```diff
@@ -1,188 +1,108 @@
-      }
-      return;
+      return null;
     };
   }
 
   public static dateTimeValidator(): ValidatorFn {
     return (control: AbstractControl): ValidationErrors | null => {
-      if (control.value === null || control.value === '') {
-        return;
+      if (this.isEmpty(control.value)) {
+        return null;
       }
+      const value = control.value.toString().replace('T', ' ').replace('.000', '');
+      const isValidDateTime = moment(value, 'YYYY-MM-DD HH:mm:ss', true).isValid();
       const [datePart, timePart] = control.value.toString().split('T');
-      const dateValues = datePart.split('-');
-      if (dateValues) {
-        if (dateValues[0]) {
-          const number = Number(dateValues[0]);
-          if (number === 0 || dateValues[0].length < 4) {
-            return { datetime: true, year: true, valid: false };
-          }
-        } else {
-          return { datetime: true, year: true };
-        }
-        if (dateValues[1]) {
-          const number = Number(dateValues[1]);
-          if (number === 0 || number > 12) {
-            return { datetime: true, month: true, valid: false };
-          }
-        } else {
-          return { datetime: true, month: true };
-        }
-        if (dateValues[2]) {
-          const number = Number(dateValues[2]);
-          if (number === 0 || number > 31) {
-            return { datetime: true, day: true, valid: false };
-          }
-        } else {
-          return { datetime: true, day: true };
-        }
-        if (timePart) {
-          const timeValues = timePart.split(':');
-          if (timeValues[0]) {
-            const number = Number(timeValues[0]);
-            if (number > 23) {
-              return { datetime: true, hour: true, valid: false };
-            }
-          } else {
-            return { datetime: true, hour: true };
-          }
-          if (timeValues[1]) {
-            const number = Number(timeValues[1]);
-            if (number > 59) {
-              return { datetime: true, minute: true, valid: false };
-            }
-          } else {
-            return { datetime: true, minute: true };
-          }
-          if (timeValues[2]) {
-            const number = Number(timeValues[2]);
-            if (number > 59) {
-              return { datetime: true, second: true, valid: false };
-            }
-          } else {
-            return { datetime: true, second: true };
-          }
-        }
-      }
-      const momentValue = control.value.toString().replace('T', ' ').replace('.000', '');
-      if (!moment(momentValue, 'YYYY-MM-DD HH:mm:ss', true).isValid()) {
-        return { datetime: true };
+      const [year, month, day] = datePart.split('-').map(Number);
+      const [hour, minute, second] = (timePart || '').split(':').map(Number);
+
+      if (!isValidDateTime || year === 0 || month === 0 || month > 12 || day === 0 || day > 31 ||
+          hour > 23 || minute > 59 || (second !== undefined && second > 59)) {
```

Comment:
Same comment as above - it doesn' return the success values any more

Reply:
Please see my comments above. It's working as intended.